From a7fd08ae367e22b3db28726e9bab7bc6f4b2040d Mon Sep 17 00:00:00 2001
From: OpenDroid <opendroid@localhost>
Date: Wed, 19 Aug 2026 15:11:28 +0000
Subject: [PATCH] rtl8852cu: split common objects

---
 Makefile  | 4 ++--
 common.mk | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index eea6243..30b64ba 100644
--- a/Makefile
+++ b/Makefile
@@ -934,8 +934,8 @@ endif
 endif
 
 obj-$(CONFIG_RTL8852CU) := $(MODULE_NAME).o
-obj-$(CONFIG_RTL8852CU) := $(MODULE_NAME).o
-$(MODULE_NAME)-y = $(OBJS)
+$(MODULE_NAME)-y = $(OBJS_COMMON)
+$(MODULE_NAME)-y += $(OBJS)
 
 ############# MEMORY MANAGMENT #############
 ifneq ($(CONFIG_RTKM), n)
diff --git a/common.mk b/common.mk
index d98f609..6aa0265 100644
--- a/common.mk
+++ b/common.mk
@@ -160,4 +160,4 @@ ifeq ($(CONFIG_RTW_MBO), y)
 _CORE_FILES +=	core/rtw_mbo.o core/rtw_ft.o core/rtw_wnm.o
 endif
 
-OBJS += $(_OS_INTFS_FILES) $(_CORE_FILES)
+OBJS_COMMON += $(_OS_INTFS_FILES) $(_CORE_FILES)
-- 
2.43.0

